src/expressions/range-expr.md

```diff
@@ -1,7 +1,35 @@
 # Range expressions
 
+> **<sup>Syntax</sup>**  
+> _RangeExpression_ :  
+> &nbsp;&nbsp; &nbsp;&nbsp; _RangeExpr_  
+> &nbsp;&nbsp; | _RangeFromExpr_  
+> &nbsp;&nbsp; | _RangeToExpr_  
+> &nbsp;&nbsp; | _RangeFullExpr_  
+>  
+> _RangeExpr_ :  
+> &nbsp;&nbsp; [_Expression_] `..` [_Expression_]  
+>  
+> _RangeFromExpr_ :  
+> &nbsp;&nbsp; [_Expression_] `..`  
+>  
+> _RangeToExpr_ :  
+> &nbsp;&nbsp; `..` [_Expression_]  
+>  
+> _RangeFullExpr_ :  
+> &nbsp;&nbsp; `..`  
+
 The `..` operator will construct an object of one of the `std::ops::Range` (or
-`core::ops::Range`) variants.
+`core::ops::Range`) variants, according to the following table:
+
+| Production             | Syntax        | Type                         | Range                 |
+|------------------------|---------------|------------------------------|-----------------------|
+| _RangeExpr_            | start`..`end  | [std::ops::Range]            | start &le; x &lt; end |
```

Comment:
Might be worth using &lt;pre&gt; tags to preserve the formatting in the Range column?

Reply:
Do you mean to keep the &le;, x and &lt; aligned (like they are in the source code)?

Reply:
Yes.